File: C#/BluffinMuffin.Protocol.DataTypes/Options/BlindOptions.cs
using BluffinMuffin.Protocol.DataTypes.Attributes;
using BluffinMuffin.Protocol.DataTypes.Enums;

namespace BluffinMuffin.Protocol.DataTypes.Options
{
    /// <summary>
    /// The type of blinds the table uses (none, blinds, antes) 
    /// </summary>
    public abstract class BlindOptions : IOption<BlindTypeEnum>
    {
        /// <summary>
        /// The Money unit. Should always be equal to the moneyUnit of the table.
        /// </summary>
        [ExampleValue(10)]
        public int MoneyUnit { get; set; }
        /// <summary>
        /// The type of blinds used for the table
        /// </summary>
        public abstract BlindTypeEnum OptionType { get; }
    }
}

File: Python/BluffinMuffin.Protocol/DataTypes/BlindOptions.py
from DataTypes.Enums.BlindTypeEnum import BlindTypeEnum

__author__ = 'ericm'
_project_ = 'BluffinMuffin.Protocol'


class BlindOptions:
    def __init__(self, obj):
        self.money_unit = obj['MoneyUnit']
        self.option_type = BlindTypeEnum.parse(obj['OptionType'])

    def __str__(self):
        return BlindTypeEnum.to_string(self.option_type)

class BlindOptionsAnte(BlindOptions):
    def __init__(self, obj):
        super().__init__(obj)
        self.ante_amount = self.money_unit

    def __str__(self):
        return '{0} ({1})'.format(super().__str__(), self.ante_amount)

class BlindOptionsBlinds(BlindOptions):
    def __init__(self, obj):
        super().__init__(obj)
        self.big_blind_amount = self.money_unit
        self.small_blind_amount = self.money_unit // 2

    def __str__(self):
        return '{0} ({1}/{2})'.format(super().__str__(), self.small_blind_amount, self.big_blind_amount)

class BlindOptionsNone(BlindOptions):
    def __init__(self, obj):
        super().__init__(obj)

class BlindOptionsDecoder():
    def decode(obj):
        type = BlindTypeEnum.parse(obj['OptionType'])
        if type == BlindTypeEnum.Antes:
            return BlindOptionsAnte(obj)
        if type == BlindTypeEnum.Blinds:
            return BlindOptionsBlinds(obj)
        if type == BlindTypeEnum.Nothing:
            return BlindOptionsNone(obj)
        return None
